shared/nav.desktop.js

```diff
@@ -63,47 +55,57 @@ class Nav extends Component {
     // Restartup when we connect online.
     // If you startup while offline, you'll stay in an errored state
     window.addEventListener('online', this.props.startup)
+
+    this._renderedActiveTab = null // the last tab we actually drew
   }
 
   _handleTabsChange (e, key, payload) {
     this.props.switchTab(e)
   }
 
-  componentWillReceiveProps (nextProps) {
```

Comment:
this was getting really messy so i just switched to using a private method with some more bookkeeping